ts/blueprints/ember-cli-typescript/index.js

```diff
@@ -3,16 +3,36 @@
 const fs = require('fs');
 const path = require('path');
 
-const APP_DECLARATIONS = `
-import Ember from 'ember';
+const APP_DECLARATIONS = `import Ember from 'ember';
 
 declare global {
   interface Array<T> extends Ember.ArrayPrototypeExtensions<T> {}
   // interface Function extends Ember.FunctionPrototypeExtensions {}
 }
 
-export {};
+export {};`;
+
+/**
+ * @param {string} projectName
+ * @param {'classic' | 'pods' | 'mu'} layout
+ */
+function buildTemplateDeclarations(projectName, layout) {
+  const comment = '// Types for compiled templates';
+  const moduleBody = `
+  import { TemplateFactory } from 'htmlbars-inline-precompile';
+  const tmpl: TemplateFactory;
+  export default tmpl;
 `;
+  switch (layout) {
+    case 'classic': return `${comment}
+declare module '${projectName}/templates/*' { ${moduleBody}}`;
+    case 'pods': return `${comment}
+declare module '${projectName}/components/*/template' { ${moduleBody}}`;
```

Comment:
I think this needs to account for `podModulePrefix`—it will also miss any route templates that are defined. Given how loose things are with pods, the simplest approach might just be `declare module '${projectName}/*/template' {`

Reply:
Thanks! I'll go with your "simplest approach" idea